core/api.go

```diff
@@ -243,6 +243,28 @@ type page struct {
 	LastPage bool         `json:"last_page"`
 }
 
+func (a *API) authnHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		req = (&authn.API{
+			Tokens:           a.accessTokens,
+			TokenMap:         make(map[string]authn.TokenResult),
+			NetworkRPCPrefix: networkRPCPrefix,
+		}).Authenticate(req)
+
+		handler.ServeHTTP(rw, req)
+	})
+}
+
+func authzHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		if authz.Authorized(req.Context()) {
+			handler.ServeHTTP(rw, req)
+		} else {
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

This suggests this should look like:

```go
if !authz.Authorized(req.Context()) {
	errorFormatter.Write(req.Context(), rw, errNotAuthenticated)
	return
}
handler.ServeHTTP(rw, req)
```